### PR TITLE
tesseract: fix cross-build to ios + bump deps + modernize

### PR DESCRIPTION
**Pull request by SpaceIm**
Specify library name and version:  **lib/1.0**

Actually cross-build to iOS is still broken due to https://github.com/conan-io/conan/issues/9202, but I can cross-build tesseract locally with this fix in conan client https://github.com/conan-io/conan/issues/9202#issuecomment-881762492

---

- [ ] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [ ] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [ ] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [ ] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SpaceIm** [comment]
Upstream PR: https://github.com/tesseract-ocr/tesseract/pull/3500

**Review by madebr — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

File: recipes/tesseract/all/patches/0001-install-bundle.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -537,7 +537,7 @@ get_target_property(tesseract_VERSION libtesseract VERSION)
+ get_target_property(tesseract_OUTPUT_NAME libtesseract OUTPUT_NAME)
+ configure_file(tesseract.pc.cmake ${CMAKE_CURRENT_BINARY_DIR}/tesseract.pc @ONLY)
+ install(FILES ${CMAKE_CURRENT_BINARY_DIR}/tesseract.pc DESTINATION lib/pkgconfig)
+-install(TARGETS tesseract RUNTIME DESTINATION bin LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
++install(TARGETS tesseract DESTINATION bin)
```
Comment (SSE4) [inline recipes/tesseract/all/patches/0001-install-bundle.patch:8]: so, aren't `RUNTIME`, `LIBRARY` and `ARCHIVE` destination needed to be set?
if so, why?
I think it's important to have libraries installed to the proper destination

Comment (SpaceIm) [reply recipes/tesseract/all/patches/0001-install-bundle.patch:8]: tesseract target is an executable. The lib is installed in another command.

Comment (SSE4) [reply recipes/tesseract/all/patches/0001-install-bundle.patch:8]: if it's an executable, it needs at least `RUNTIME DESTINATION`, right? (and maybe `BUNDLE DESTINATION` for OSX)

Comment (SpaceIm) [reply recipes/tesseract/all/patches/0001-install-bundle.patch:8]: DESTINATION alone set the destination for all types.

**Review by SSE4 — APPROVED** [review]

File: recipes/tesseract/all/patches/0001-install-bundle.patch (hunk 1 above)
Comment (SpaceIm) [reply recipes/tesseract/all/patches/0001-install-bundle.patch:8]: And from CMake documentation: https://cmake.org/cmake/help/latest/command/install.html

> For each of these arguments given, the arguments following them only apply to the target or file type specified in the argument. If none is given, the installation properties apply to all target types. If only one is given then only targets of that type will be installed (which can be used to install just a DLL or just an import library.)

**Pull request merged**